File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s) < 2:
            return len(s)

        seen = set()
        l, r = 0, 0
        res = 0

        while r < len(s):
            if s[r] in seen:
                while l < r and s[r] in seen:
                    seen.remove(s[l])
                    l += 1
            seen.add(s[r])
            res = max(res, r - l + 1)
            r += 1
        
        return max(res, r - l)

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.js
class Solution {
    /**
     * @param {number[]} prices
     * @return {number}
     */
    maxProfit(prices) {
        let res = 0, lowest = prices[0];

        for (let price of prices) {
            lowest = Math.min(lowest, price);
            res = Math.max(res, price - lowest);
        }

        return res;
    }
}
